x-pack/plugins/alerting/public/pages/maintenance_windows/components/submit_button.tsx

```diff
@@ -11,7 +11,11 @@ import { EuiButton } from '@elastic/eui';
 import { useFormContext } from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
 import * as i18n from '../translations';
 
-export const SubmitButton: React.FC = React.memo(() => {
+interface SubmitButtonProps {
+  editMode?: boolean;
+}
+
+export const SubmitButton: React.FC<SubmitButtonProps> = React.memo((editMode) => {
```

Comment:
I noticed when I hit submit, we prematurely re-enable the submit button. It doesn't look too bad here but if the connection speed of the user was a bit slower then I can imagine this being a bit more jarring. I'm guessing this is because we're checking `isSubmitting` which isn't accounting for the entire duration of HTTP request, so I think we should check the `isLoading` flag coming from the react query mutation instead. 


https://user-images.githubusercontent.com/74562234/232615421-434ca1ff-c335-41d2-ba23-7190b832554e.mp4

Reply:
Resolved in this commit 668904dcc0023fa6f7698689bda52a0e2541cc70